src/index.js

```diff
@@ -54,54 +54,58 @@ ipcMain.on('setup', async () => {
     mainWindow.webContents.send('setup-start');
     const setUpComplete = await setup();
 
-    if (!setUpComplete.status) {
-      console.error('Setup Failed: No Wild Encounter Detected On-Screen', setUpComplete.error);
-      mainWindow.webContents.send('setup-failed', setUpComplete);
-    } else {
-      console.log('Setup complete');
-      huntingWindow = setUpComplete.window;
-      huntingDisplayId = setUpComplete.displayId;
-      mainWindow.webContents.send('setup-complete', setUpComplete);
-      huntSession.setUpWindow(huntingWindow, huntingDisplayId);
+    if (!setUpComplete.status || !setUpComplete.window || !setUpComplete.displayId) {
+      throw new Error(setUpComplete.error);
```

Comment:
A couple of thoughts. 

1. I'm curious why you've introduced other failure conditions to this.
2. Where does `setUpComplete.error` come from and why are we using it here? Is `setUpComplete` a type which sometimes has `error: string` IFF `status === false`?
3. Notwithstanding structural implications of the above questions, I'd encourage you to handle separate error conditions separately also, unless there's a compelling reason not to.

E.g.,

```javascript
    if (errCondition1) {
      throw new Error("Failed because of reason 1");
    }
    if (errCondition2) {
      throw new Error("Failed because of reason 2");
    }
    ...
```

Reply:
Looking through the functions in `helper.js` I notice that `setup` and `processScreen` both do 

```javascript
try { ... } catch { return ...; } }
```

I wonder whether that's the best way to deal with errors. E.g., one of the smaller functions sometimes returns a value and sometimes returns `null`. Maybe that function should throw an exception instead? And then maybe `processScreen` should just let that exception bubble up? And then maybe this function should let that exception bubble up? Maybe exceptions' ability to halt a bunch of stack frames could be an advantage here and not a disadvantage?


Separately, if you end up migrating to TS, I think you'd get a lot of milage out of using some kind of `Result` type. E.g., consider this code:


```typescript
type Result<T, E> = { ok: true, value: T } | { ok: false, error: E};

function inner(): Result<number, Error> {
    let res: Result<number, Error>;

    res = { ok: true, value: 123 };                 // Good
    res = { ok: false, error: new Error("uh oh") }; // Good

    res = { ok: false, value: 123 };                // Error: Not assignable to type
    res = { ok: true, error: "wtf" };               // Error: Not assignable to type

    return res;
}


function outer() {
    let res = inner();
    console.log(res.value);                     // Error: Property does not exist
    if (!res.ok) { console.error(res.error); }  // Good
    if (res.ok) { console.log(res.value); }     // Good
}
```

Notice that:
1. You aren't allowed to write invalid modes to the `Result` type (ok + error, or not ok + value)
2. You aren't allowed to consider invalid modes of any given `Result` type (using the value without checking to see if it's ok)